pkg/daemon/ceph/osd/daemon.go

```diff
@@ -143,26 +145,13 @@ func Run(context *clusterd.Context, agent *OsdAgent, done chan struct{}) error {
 			logger.Warningf("failed to clean up node resources, they may need to be cleaned up manually: %+v", err)
 		}
 	}
-
+	osds := append(deviceOsds, dirOsds...)
 	// orchestration is completed, update the status
-	status = oposd.OrchestrationStatus{Status: oposd.OrchestrationStatusCompleted}
+	status = oposd.OrchestrationStatus{OSDs: osds, Status: oposd.OrchestrationStatusCompleted}
```

Comment:
A few comments on this `Run` method:
- The device and directory discovery should be triggered by the rook agent instead of the osd. Hopefully it was clear from the comments on the design doc. 
- The osd pod that triggers this discovery is no longer needed. We can get rid of the osd daemonset.
- This method can be simplified. All it needs to do is discovery devices and directories, then update the configmap. It doesn't need to do things like generate config files.